### PR TITLE
fix: crash using `powerMonitor` before ready event

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/40694.
Refs https://github.com/electron/electron/pull/37937.

This PR fixes a crash resultant in trying to listen to power-related events before the `ready` event is emitted on Linux. This was happening because the `PowerObserverLinux` ctor called `bluez::BluezDBusThreadManager::Get()`, which assumed that `bluez::BluezDBusThreadManager::Initialize()` had already been called and `CHECK`ed if it hadn't been. However, our existing logic did not call `bluez::BluezDBusThreadManager::Initialize()` until `ElectronBrowserMainParts::PostCreateMainMessageLoop()`, which meant that trying to use:

```js
  powerMonitor.on('suspend', () => {
    console.log('suspend')
  });

  powerMonitor.on('resume', () => {
    console.log('resume')
  });
```

before the `ready` event would immediately die. I chose to fix this by calling `bluez::BluezDBusThreadManager::Initialize()` as soon as it was needed, and adding a check to ensure that it was not initialized more than once. Alternately, we can bring back the `ready` event requirement.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fixed a crash resultant from trying to listen to power-related events before the `ready` event was emitted on Linux.

### CODE REVIEW

**Review by nornagon — APPROVED** [review]
lazy initialize seems like a good approach to me 👍🏻

**Review by VerteDinde — APPROVED** [review]

File: shell/browser/electron_browser_main_parts.cc [hunk 1]
```diff
@@ -386,7 +387,8 @@ void ElectronBrowserMainParts::PostDestroyThreads() {
 #endif
 #if BUILDFLAG(IS_LINUX)
   device::BluetoothAdapterFactory::Shutdown();
-  bluez::DBusBluezManagerWrapperLinux::Shutdown();
+  bluez::BluezDBusManager::Shutdown();
```
Comment (ckerr) [inline shell/browser/electron_browser_main_parts.cc:390]: This function `CHECK()`s that `bluez::BluezDBusManager::Initialize()` has been called first, so  I _think_ we need to wrap this with `if (bluez::BluezDBusManager::IsInitialized())`?

```suggestion
  if (bluez::BluezDBusManager::IsInitialized())
    bluez::BluezDBusManager::Shutdown();
```

**Review by ckerr — CHANGES_REQUESTED** [review]

**Review by ckerr — APPROVED** [review]

**Review by deepak1556 — COMMENTED** [review]
Minor nit: wouldn't it be better to continue using the wrapper class to avoid missing any additional bluez initialization or shutdown paths that might get added in future ?

```
if (!bluez::BluezDBusManager::IsInitialized())
  bluez::DBusBluezManagerWrapperLinux::Initialize();
```

**Comment by codebytere** [comment]
@deepak1556 done!

**Review by deepak1556 — APPROVED** [review]
Thanks!

**Comment by codebytere** [comment]
Failing test unrelated: 
> reporting api sends a report for an intervention

**Pull request merged**